# Functional annotations
```
cd /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline
```
Getting InterproScan (you may copy the the tarball and md5 file from /fs/ess/PAS2303/HCS7004_Files/Annotation)
```
tee -a GetInterProScan.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name InterProScan
#SBATCH --time=03:00:00
#SBATCH --ntasks=1
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303
# Training for gene prediction and functional annotation

```
cd /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline
```

# Batch script for Funannotate training
```
tee -a Training.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Funannotate_Training
#SBATCH --time=03:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline
# Running Funannotate
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate train \
--cpus 28 -i /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Genome_Preparation/Nanopore_Illumina_clean_sorted_masked.fasta -o Peltaster_fructicola \
--left \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115198_1_restored_clean_renamed.fastq.gz \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115199_1_restored_clean_renamed.fastq.gz \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115200_1_restored_clean_renamed.fastq.gz \
--right \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115198_2_restored_clean_renamed.fastq.gz \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115199_2_restored_clean_renamed.fastq.gz \
/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Preprocessing/trimming/SRR8115200_2_restored_clean_renamed.fastq.gz \
--stranded RF --jaccard_clip --species "Peltaster fructicola" --strain LNHT1506
EOF
```
Then, we send it to the scheduler:
##
```
sbatch Training.sh
squeue -u kellymikhail
```
##Slurm Job_id=21102742 Name=Funannotate_Training Began, Queued time 00:00:00

Download databases for prediction (you can do this while the training takes place)
```
mkdir -p data/database
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate setup -i all -w -b all --database data/database
# adding repeats
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate setup -i repeats --database data/database -w -f --database data/database
# adding BUSCO database
cd data/database
wget https://busco-data.ezlab.org/v4/data/lineages/dothideomycetes_odb10.2020-08-05.tar.gz
tar -pxvzf dothideomycetes_odb10.2020-08-05.tar.gz
# Updating databases, just in case
cd ../..
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate setup --update --database data/database
```
# Batch script for Funannotate prediction (wait for the training batch job ends and you have all the databses installed)
squeue -u kellymikhail
```
tee -a Prediction.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Funannotate_Prediction
#SBATCH --time=01:30:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline
# Making databases available for funannotate
export APPTAINER_BIND="/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline/data/database:/data/database"
module load perl/5.26.1
# Running Funannotate
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate predict -i /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Genome_Preparation/Nanopore_Illumina_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/gmes_linux_64_4
EOF
```
Then, we send it to the scheduler:
```
sbatch Prediction.sh
squeue -u kellymikhail
```
## Sunday night at this point 8:11PM
# Identify UTRs and update
```
tee -a Update.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Update
#SBATCH --time=03:00:00
#SBATCH --ntasks=2
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline
# Making databases available for funannotate
export APPTAINER_BIND="/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline/data/database:/data/database"
module load perl/5.26.1
# Running Funannotate
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate update -i Peltaster_fructicola --cpus 2
EOF
```
Then, we send it to the scheduler:
```
sbatch Update.sh
squeue -u kellymikhail
```
# Add the data prediction generated to the database
```
mkdir data/database/trained_species
export APPTAINER_BIND="/fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Funannotate_pipeline/data/database/trained_species:/data/database/trained_species"
apptainer run /fs/ess/scratch/PAS2303/Kelly/Lab7/Annotation/Software/Funannotate.sif funannotate species -s peltaster_fructicola_lnht1506 -a Peltaster_fructicola/predict_results/peltaster_fructicola_lnht1506.parameters.json
```

# Working directory
cd /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline
# Downloading InterProScan and its md5 file
wget https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.59-91.0/interproscan-5.59-91.0-64-bit.tar.gz
wget https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.59-91.0/interproscan-5.59-91.0-64-bit.tar.gz.md5
EOF
```
Then, we send it to the scheduler:
```
sbatch GetInterProScan.sh
squeue -u your_OSC_ID
```
Checking integrity and installation
```
md5sum -c interproscan-5.59-91.0-64-bit.tar.gz.md5
# Decompressing
tar -pxvzf interproscan-5.59-91.0-64-bit.tar.gz
rm interproscan-5.59-91.0-64-bit.tar.gz
# Installing
module load python/3.9-2022.05
module load perl/5.26.1
module load java/12.0.2
cd interproscan-5.59-91.0
python3 setup.py interproscan.properties
# Testing
./interproscan.sh -i test_all_appl.fasta -f tsv -dp
./interproscan.sh -i test_all_appl.fasta -f tsv
```
# Batch script for running InterProScan
```
tee -a runIPRSCAN.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name InterProScan
#SBATCH --time=01:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline
# Running InterProScan locally
apptainer run --bind /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline/data/database:/data/database /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Software/Funannotate.sif funannotate iprscan -i Peltaster_fructicola -m local -c 28 --iprscan_path /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline/interproscan-5.59-91.0
EOF
```
Then, we send it to the scheduler:
```
sbatch runIPRSCAN.sh
squeue -u your_OSC_ID
```

# Annotate with Phobius (transmembrane topology and signal peptide predictor)
```
tee -a runPhobius.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Phobius
#SBATCH --time=01:00:00
#SBATCH --ntasks=1
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline
# Remote search in phobius
apptainer run /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Software/Funannotate.sif funannotate remote -m phobius -e your_osu_email@osu.edu -i Peltaster_fructicola
EOF
```
Then, we send it to the scheduler:
```
sbatch runPhobius.sh
squeue -u your_OSC_ID
```

# Perform final functional annotation procedure
```
tee -a FuncAnnot.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Annotation
#SBATCH --time=03:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Annotation/Funannotate/Functional_annotation
# Running functional annotation
apptainer run --bind /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Funannotate_pipeline/data/database:/data/database /fs/ess/scratch/PAS2303/your_OSC_ID/Annotation/Software/Funannotate.sif funannotate annotate -i Peltaster_fructicola --busco_db dothideomycetes_odb10 --cpus 28 --force
EOF
```
Then, we send it to the scheduler:
```
sbatch FuncAnnot.sh
squeue -u your_OSC_ID
```
